src/qpmodel.jl

```diff
@@ -131,29 +130,26 @@ function cons(qp::AbstractQuadraticModel, x :: AbstractVector)
     cons!(qp, x, c)
 end
 
-function cons!(qp :: AbstractQuadraticModel, x :: AbstractVector, c :: AbstractVector)
+function NLPModels.cons!(qp :: AbstractQuadraticModel, x :: AbstractVector, c :: AbstractVector)
     mul!(c, qp.data.A, x)
     qp.counters.neval_jprod += 1
     c
 end
 
-function hprod(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...)
-    @closure v -> begin
-        qp.counters.neval_hprod += 1
-        qp.data.opH * v
-    end
+function NLPModels.hprod!(qp :: AbstractQuadraticModel, x :: AbstractVector, v :: AbstractVector, Av :: AbstractVector; kwargs...)
+    qp.counters.neval_hprod += 1
+    Av .= qp.data.opH * v
+    return Av
 end
 
-function jprod(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...)
-    @closure v -> begin
-        qp.counters.neval_jprod += 1
-        qp.data.A * v
-    end
+function NLPModels.jprod!(qp :: AbstractQuadraticModel, x :: AbstractVector, v :: AbstractVector, Av :: AbstractVector; kwargs...)
+    qp.counters.neval_jprod += 1
+    Av .= qp.data.A * v
```

Comment:
If we assume that `qp.data.A` is an `AbstractMatrix`, we should be able to use `mul!()`.

Reply:
Done.